slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -407,6 +431,10 @@ void SLAMServiceImpl::CreateMap() {
     cartographer::transform::Rigid3d tmp_global_pose =
         cartographer::transform::Rigid3d();
     while (file != "") {
+        if (file.find(".pcd") == std::string::npos) {
+            file = GetNextDataFile();
+            continue;
+        }
```

Comment:
This ensures that if there's e.g. a DS_Store file present, that we're not trying to read it in.

Reply:
This check should be included in GetNextDataFile

Reply:
Do you mean that checking if the file is a pcd file or not should be implemented within `GetNextDataFile`?

Reply:
I personally think so since it will have to be run anytime `GetNextDataFile` is run and we are running additional checks there, nice to have them all in one space.